### PR TITLE
Create a collection type for signing keys

### PR DESCRIPTION
**Pull request by poljar**

### CODE REVIEW

File: crates/matrix-sdk-crypto/src/identities/user.rs [hunk 1]
```diff
@@ -411,7 +411,7 @@ impl MasterPubkey {
     /// # Arguments
     ///
     /// * `key_id` - The id of the key that should be fetched.
-    pub fn get_key(&self, key_id: &DeviceKeyId) -> Option<&SigningKey> {
+    pub fn get_key(&self, key_id: &OwnedDeviceKeyId) -> Option<&SigningKey> {
```
Comment (jplatte) [inline crates/matrix-sdk-crypto/src/identities/user.rs:414]: A reference to an owned ID type? This seems wrong.

File: crates/matrix-sdk-crypto/src/types/mod.rs [hunk 2]
```diff
@@ -229,6 +230,121 @@ impl Serialize for Signatures {
...
+
+    /// Insert a `SigningKey` into the collection.
+    pub fn insert(&mut self, key_id: T, key: SigningKey) -> Option<SigningKey> {
+        self.0.insert(key_id, key)
+    }
+
+    /// Get a `SigningKey` with the given `DeviceKeyId`.
+    pub fn get(&self, key_id: &T) -> Option<&SigningKey> {
+        self.0.get(key_id)
+    }
```
Comment (jplatte) [inline crates/matrix-sdk-crypto/src/types/mod.rs:251]: I think this method's signature should mirror the underlying map's one (using `Borrow`), maybe then the `&OwnedDeviceId` won't be needed. If it still is, I'll look into adding any missing `Borrow` implementations in Ruma.

**Review by jplatte — COMMENTED** [review]
First comment ends in a period, it probably shouldn't.

File: crates/matrix-sdk-crypto/src/types/mod.rs (hunk 2 above)
Comment (poljar) [reply crates/matrix-sdk-crypto/src/types/mod.rs]: Right, that does indeed work: https://github.com/matrix-org/matrix-rust-sdk/pull/943/commits/7665d0bad7e09fe9ca2b299032dad928b4f338d5.

File: crates/matrix-sdk-crypto/src/types/mod.rs [hunk 3]
```diff
@@ -246,7 +249,11 @@ impl<T: Ord> SigningKeys<T> {
     }
 
     /// Get a `SigningKey` with the given `DeviceKeyId`.
-    pub fn get(&self, key_id: &T) -> Option<&SigningKey> {
+    pub fn get<Q: ?Sized>(&self, key_id: &Q) -> Option<&SigningKey>
+    where
+        T: Borrow<Q> + Ord,
```
Comment (jplatte) [inline crates/matrix-sdk-crypto/src/types/mod.rs:254]: The `Ord` constraint here is already specified at the `impl` level.

**Review by jplatte — APPROVED** [review]
Other than the redundant trait bound, this looks good to me.

**Pull request merged**